src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/appendoverwrite/middleware/commit/BrAPIObservationCommit.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.breedinginsight.brapps.importer.services.processors.experiment.appendoverwrite.middleware.commit;
+
+import io.micronaut.context.annotation.Prototype;
+import lombok.extern.slf4j.Slf4j;
+import org.brapi.client.v2.model.exceptions.ApiException;
+import org.brapi.v2.model.pheno.BrAPIObservation;
+import org.breedinginsight.brapps.importer.services.processors.experiment.appendoverwrite.factory.action.BrAPICreationFactory;
+import org.breedinginsight.brapps.importer.services.processors.experiment.appendoverwrite.factory.action.WorkflowCreation;
+import org.breedinginsight.brapps.importer.services.processors.experiment.appendoverwrite.factory.action.BrAPIUpdateFactory;
+import org.breedinginsight.brapps.importer.services.processors.experiment.appendoverwrite.factory.action.WorkflowUpdate;
+import org.breedinginsight.brapps.importer.services.processors.experiment.appendoverwrite.model.AppendOverwriteMiddleware;
+import org.breedinginsight.brapps.importer.services.processors.experiment.appendoverwrite.model.AppendOverwriteMiddlewareContext;
+import org.breedinginsight.brapps.importer.services.processors.experiment.appendoverwrite.model.MiddlewareException;
+import org.breedinginsight.services.exceptions.DoesNotExistException;
+import org.breedinginsight.services.exceptions.MissingRequiredInfoException;
+import org.breedinginsight.services.exceptions.UnprocessableEntityException;
+import org.breedinginsight.utilities.Utilities;
+
+import javax.inject.Inject;
+import java.util.Optional;
+
+@Slf4j
+@Prototype
+public class BrAPIObservationCommit extends AppendOverwriteMiddleware {
+    private final BrAPICreationFactory brAPICreationFactory;
+    private final BrAPIUpdateFactory brAPIUpdateFactory;
+    private WorkflowCreation<BrAPIObservation> brAPIObservationCreation;
+    private WorkflowUpdate<BrAPIObservation> brAPIObservationUpdate;
+    private Optional<WorkflowCreation.BrAPICreationState> createdBrAPIObservations;
```

Comment:
Both `WorkfolwCreation` and `BrAPICreationState` are parametrized classes, but both are used as raw classes.
Is this OK? (it compiles without problem).

Reply:
Yep it compiles fine, but should be fixed. Turns out the BrAPIState interface needed to be parameterized, then the type parameters worked. I updated it here and the rest of the Commit classes and other actions, read and update.